contrib/coredns/policy/setup.go

```diff
@@ -38,19 +37,11 @@ func setup(c *caddy.Controller) error {
 			return plugin.Error("policy", err)
 		}
 
-		if mh := dnsserver.GetConfig(c).Handler("prometheus"); mh != nil {
-			if m, ok := mh.(*metrics.Metrics); ok && policyPlugin.SetupMetrics() {
-				metricsOnce.Do(func() {
-					m.MustRegister(policyPlugin.attrGauges.pgv)
-				})
-			}
-		}
-		return nil
+		return policyPlugin.SetupMetrics(c)
 	})
 
 	c.OnShutdown(func() error {
 		policyPlugin.closeConn()
-		policyPlugin.attrGauges.Stop()
```

Comment:
I can see why we don't call `AttrGuage.Stop()` from here anymore.

However, it now looks like the `AttrGuage.Stop()` is not used at all, except in UTs.

Is that intentional?

Reply:
That's right, now the global AttrGauge (if started) will run until the coredns process is terminated

Reply:
Thanks for the confirmation!

I'm going to approve it. I suggest you make a comment in the code (maybe where Start() is called) to that effect so it's clear.